cmake/modules/extensions.cmake

```diff
@@ -718,6 +718,222 @@ function(generate_inc_file_for_target
+  set(multi_args "BINARY_FILES;SYMBOL_NAMES")
+  cmake_parse_arguments(PARSE_ARGV 1 BINARY_EMBED "${options}"
+    "${single_args};${opt_single_args}" "${multi_args}")
+
+  if(BINARY_EMBED_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "generate_binary_embed_inc_file_for_target(${ARGV0} ...) given unknown "
+                        "arguments: ${BINARY_EMBED_UNPARSED_ARGUMENTS}"
+    )
+  endif()
+
+  foreach(arg ${single_args};${multi_args})
+    if(NOT DEFINED BINARY_EMBED_${arg})
+      message(FATAL_ERROR "generate_binary_embed_inc_file_for_target(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  list(LENGTH BINARY_EMBED_BINARY_FILES bin_files_count)
+  list(LENGTH BINARY_EMBED_SYMBOL_NAMES symbol_names_count)
+  if(NOT symbol_names_count EQUAL bin_files_count)
+      message(FATAL_ERROR "Number of specified binary files (${bin_files_count}) is not the same "
+              "as number of the corresponding symbols names (${symbol_names_count})")
+  endif()
+
+  set(generated_file ${BINARY_EMBED_OUTPUT})
+  set(binary_files ${BINARY_EMBED_BINARY_FILES})
+  set(symbol_names ${BINARY_EMBED_SYMBOL_NAMES})
+  set(section ${BINARY_EMBED_SECTION})
+
+  if(NOT DEFINED BINARY_EMBED_READONLY)
+    set(readonly yes)
+  else()
+    set(readonly ${BINARY_EMBED_READONLY})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_ALIGN)
+    set(align 4)
+  else()
+    set(align ${BINARY_EMBED_ALIGN})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_OFFSET)
+    set(offset 0)
+  else()
+    set(offset ${BINARY_EMBED_OFFSET})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_LENGTH)
+    set(length -1)
+  else()
+    set(length ${BINARY_EMBED_LENGTH})
+  endif()
+
+  if(NOT DEFINED BINARY_EMBED_COMPRESS)
+    set(compress no)
+  else()
+    set(compress ${BINARY_EMBED_COMPRESS})
+    if (compress STREQUAL gzip)
+      set(compress_param -g)
+    else()
+      message(WARNING "Unusupported compression method.")
+    endif()
+  endif()
+
+  if(DEFINED BINARY_EMBED_GZIP_MTIME)
+    if(NOT compress STREQUAL gzip)
+      if(DEFINED gzip_mtime_param)
+        message(WARNING "GZIP_MTIME specified without enabling gzip compression - ignored.")
+      endif()
+    else()
+      if (BINARY_EMBED_GZIP_MTIME STREQUAL "now")
+        set(gzip_mtime_param --gzip-mtime)
+      else()
+        set(gzip_mtime_param --gzip-mtime=${BINARY_EMBED_GZIP_MTIME})
+      endif()
+    endif()
+  endif()
+
+  # Enable use of GNU-style incbin directive, if toolchain being used supports
+  # GNU extensions and compression or partial inclusion is not requested.
+  if(CONFIG_TOOLCHAIN_SUPPORTS_GNU_EXTENSIONS AND
+    (offset EQUAL 0) AND (length EQUAL -1) AND (zip STREQUAL no))
+    set(use_incbin yes)
+  else()
+    set(use_incbin no)
+  endif()
+
+  set(generated_files ${generated_file})
+  if(use_incbin)
+    list(APPEND generated_files ${generated_file}.S)
+    zephyr_library_sources(${generated_file}.S)
```

Comment:
Why is the generated assemble file added to the current Zephyr library, even when dedicated `target` is specified ?